internal/numeric.h

```diff
@@ -127,6 +127,37 @@ VALUE rb_int_bit_length(VALUE num);
 VALUE rb_int_uminus(VALUE num);
 VALUE rb_int_comp(VALUE num);
 
+// Unified 128-bit integer structures that work with or without native support:
+struct rb_uint128_t {
+    union {
+        struct {
+            uint64_t low;
+            uint64_t high;
```

Comment:
This order seems to assume little endian.
Select by `WORDS_BIGENDIAN`, or move these to io_buffer.c if io_buffer.c specific types.